### PR TITLE
Feat framework rewrite

### PR DESCRIPTION
**Pull request by yacheng**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -129,5 +129,8 @@
     "commitizen": {
       "path": "./node_modules/cz-conventional-changelog"
     }
+  },
+  "dependencies": {
+    "weex-js-runtime": "^0.23.6"
```
Comment (yuanyan) [inline package.json]: 这个依赖是哪里使用，加在这的依赖目前都是工具使用或demo使用的

File: packages/weex-rax-framework-api/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+weex
```
Comment (yuanyan) [inline packages/weex-rax-framework-api/.gitignore]: 忽略的 weex 目录是？

File: packages/weex-rax-framework-api/src/index.js [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+      }
+    },
+    get onerror() {
+      return errorHandler;
+    },
+    // ModuleJS
+    define: __weex_define__,
+    require: __weex_require__,
+    // Weex
+    callNative: () => {},
```
Comment (yuanyan) [inline packages/weex-rax-framework-api/src/index.js]: 可以注明下代码的用途

File: packages/weex-rax-framework-api/src/index.js [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    cancelAnimationFrame,
+    setImmediate: typeof setImmediate === 'function' && setImmediate ||
+      function(fn) {
+        setTimeout(fn, 0);
+      },
+    clearImmediate: typeof setImmediate === 'function' && setImmediate ||
+      function(id) {
+        clearTimeout(id);
+      },
+    frameworkVersion: null,
```
Comment (yuanyan) [inline packages/weex-rax-framework-api/src/index.js]: 注明下用途

File: packages/weex-rax-framework-api/src/index.js [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+
+  genBuiltinModules(
+    modules,
+    ModuleFactories,
+    window
+  );
+
+  window.self = window.window = window;
+
+  console.log('Rax jsfm init window', typeof window);
```
Comment (yuanyan) [inline packages/weex-rax-framework-api/src/index.js]: 日志需要优化下

File: scripts/dist.js [hunk 6]
```diff
@@ -139,6 +139,27 @@ dist(getConfig(
       ]
     }
   ));
+
+  dist(getConfig(
+    {
+      'frameworkapi.weex': './packages/weex-rax-framework-api/src/index.js'
```
Comment (yuanyan) [inline scripts/dist.js]: framework.api.weex.js

Comment (yuanyan) [reply scripts/dist.js]: api.weex.js

File: package.json (hunk 1 above)
Comment (yacheng) [reply package.json]: 移到 devDependencies，用来测试新的运行环境

File: packages/weex-rax-framework-api/.gitignore (hunk 2 above)
Comment (yacheng) [reply packages/weex-rax-framework-api/.gitignore]: 拆分成两个文件是因为两段在不同环境中运行

**Pull request merged**